src/sentry/utils/redis.py

```diff
@@ -157,6 +159,11 @@ def call_script(client, keys, args):
         followed by the values that will be provided as ``KEYS`` and ``ARGV``
         to the script as two sequence arguments.
         """.format(path)
+        script = cache.get(client)
+        if script is None:
+            script = cache[client] = Script(client, source)
+            script.registered_client = None  # prevent circular reference
```

Comment:
Could you add more of a comment as to why we're clearing the registered client? I can't seem to see the circular reference.